src/simulator_core/entities/assets/demand_cluster.py

```diff
@@ -81,7 +215,31 @@ def write_to_output(self) -> None:
         The output list is a list of dictionaries, where each dictionary
         represents the output of its asset for a specific timestep.
         """
-        pass
+        outputs = dict()
+
+        if self.simulation_performed():
```

Comment:
What happens if simulation is not performed? I think you still need to raise an error that we do not have simulation data when you call this function.

Reply:
```suggestion
        if self.simulation_performed():
            param = val
            return {param: val
            }
        else:
            raise ValueError("Simulation data not available.")
```